components/fonts/platform/macos/core_text_font_cache.rs

```diff
@@ -60,45 +105,179 @@ impl CoreTextFontCache {
+        ))
+    }
+
+    fn add_variations_to_font(
+        font_identifier: &FontIdentifier,
+        platform_font: PlatformFont,
+        specified_variations: &[FontVariation],
+        pt_size: f64,
+    ) -> PlatformFont {
+        if specified_variations.is_empty() {
+            return platform_font;
+        }
+        let Some(variations) =
+            Self::get_variation_axis_information(font_identifier, &platform_font)
+        else {
+            return platform_font;
+        };
+
+        let mut modified_variations = false;
+        let variations: Vec<_> = variations
+            .iter()
+            .map(|variation| {
+                let value = specified_variations
+                    .iter()
+                    .find_map(|specified_variation| {
+                        if variation.tag == specified_variation.tag as i64 {
+                            Some(specified_variation.value as f64)
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or(variation.default_value)
+                    .clamp(variation.min_value, variation.max_value);
+                if value != variation.default_value {
+                    modified_variations = true;
+                }
+                FontVariation {
+                    tag: variation.tag as u32,
+                    value: value as f32,
+                }
+            })
+            .collect();
+
+        if !modified_variations {
+            return platform_font;
+        }
+
+        let cftype_variations: Vec<_> = variations
+            .iter()
+            .map(|variation| {
+                (
+                    CFNumber::from(variation.tag as i64),
+                    CFNumber::from(variation.value as f64),
+                )
+            })
+            .collect();
+        let values_dict = CFDictionary::from_CFType_pairs(&cftype_variations);
+
+        let variation_attribute =
+            unsafe { CFString::wrap_under_get_rule(kCTFontVariationAttribute) };
+        let attrs_dict = CFDictionary::from_CFType_pairs(&[(variation_attribute, values_dict)]);
+        let ct_var_font_desc = platform_font
+            .ctfont
+            .copy_descriptor()
+            .create_copy_with_attributes(attrs_dict.to_untyped())
+            .unwrap();
+
+        let ctfont = core_text::font::new_from_descriptor(&ct_var_font_desc, pt_size);
+        PlatformFont::new_with_ctfont_and_variations(ctfont, variations)
+    }
+
+    fn get_variation_axis_information(
+        font_identifier: &FontIdentifier,
+        platform_font: &PlatformFont,
+    ) -> Option<Vec<VariationAxisInformation>> {
+        let cache = VARIATION_AXIS_INFORMATION_CACHE
+            .0
+            .get_or_init(Default::default);
+        if let Some(information) = cache.read().get(font_identifier) {
+            return information.clone();
+        }
```

Comment:
Is reading the variation axes on a font really expensive enough to warrant a cache? It doesn't *look* very expensive and we don't read the variations frequently.

Reply:
Let's remove the cache for now and add it back if querying the axes becomes expensive.

Reply:
Okay. I've removed the new cache now. PTAL.